lang/src/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -361,180 +361,168 @@ internal class StaticTypeInferenceVisitorTransform(
             }
         }
 
         override fun transformExprLte(node: PartiqlAst.Expr.Lte): PartiqlAst.Expr {
-            val nAry = super.transformExprLte(node) as PartiqlAst.Expr.Lte
-            return if (operandsAreComparable(nAry.operands, "<=", nAry.metas)) {
-                transformNAry(nAry, nAry.operands) { recurseForNAryOperations(nAry, it, ::getTypeForNAryCompareOperations) }
-            } else {
-                return nAry.withStaticType(StaticType.BOOL)
+            val processedNode = super.transformExprLte(node) as PartiqlAst.Expr.Lte
+            return when (operandsAreComparable(processedNode.operands, "<=", processedNode.metas)) {
+                true -> transformNAry(processedNode, processedNode.operands) { recurseForNAryOperations(processedNode, it, ::getTypeForNAryCompareOperations) }
+                false -> processedNode.withStaticType(StaticType.BOOL)
             }
         }
 
         override fun transformExprBetween(node: PartiqlAst.Expr.Between): PartiqlAst.Expr {
-            val nAry = super.transformExprBetween(node) as PartiqlAst.Expr.Between
-            val args = listOf(nAry.value, nAry.from, nAry.to)
-            return if (operandsAreComparable(args, "BETWEEN", nAry.metas)) {
-                transformNAry(nAry, args) { recurseForNAryOperations(nAry, it, ::getTypeForNAryCompareOperations) }
-            } else {
-                return nAry.withStaticType(StaticType.BOOL)
+            val processedNode = super.transformExprBetween(node) as PartiqlAst.Expr.Between
+            val args = listOf(processedNode.value, processedNode.from, processedNode.to)
+            return when (operandsAreComparable(args, "BETWEEN", processedNode.metas)) {
+                true -> transformNAry(processedNode, args) { recurseForNAryOperations(processedNode, it, ::getTypeForNAryCompareOperations) }
+                false -> processedNode.withStaticType(StaticType.BOOL)
             }
         }
 
         // Logical NAry ops: NOT, AND, OR
         override fun transformExprNot(node: PartiqlAst.Expr.Not): PartiqlAst.Expr {
-            val nAry = super.transformExprNot(node) as PartiqlAst.Expr.Not
-            val args = listOf(nAry.expr)
-            return if (hasValidOperandTypes(args, { it is BoolType }, "NOT", nAry.metas)) {
-            transformNAry(nAry, args) { recurseForNAryOperations(nAry, it, ::getTypeForNAryLogicalOperations) }
-        } else {
-                nAry.withStaticType(StaticType.BOOL)
-            }
+            val processedNode = super.transformExprNot(node) as PartiqlAst.Expr.Not
+            val args = listOf(processedNode.expr)
+            return when (hasValidOperandTypes(args, { it is BoolType }, "NOT", processedNode.metas)) {
+            true -> transformNAry(processedNode, args) { recurseForNAryOperations(processedNode, it, ::getTypeForNAryLogicalOperations) }
+            false -> processedNode.withStaticType(StaticType.BOOL)
```

Comment:
Just an observation (won't block on this behavior by ktlint): not sure why ktlint chose to format the `when` expression this way (I would have expected the formatting that's on L414 - 417). It seems like the original `if`/`else` expression also had this formatting. Similarly for other later `when` expressions.

Reply:
I don't know either. It's wired.